wallet/chainntfns.go

```diff
@@ -107,15 +106,32 @@ func (w *Wallet) handleChainNotifications() {
 					chainClient, birthdayStore,
 				)
 				if err != nil && !waddrmgr.IsError(err, waddrmgr.ErrBirthdayBlockNotSet) {
-					panic(fmt.Errorf("unable to sanity "+
-						"check wallet birthday block: %v",
-						err))
+					log.Errorf("unable to sanity check wallet "+
+						"birthday block: %v", err)
 				}
-
-				err = w.syncWithChain(birthdayBlock)
-				if err != nil && !w.ShuttingDown() {
-					panic(fmt.Errorf("unable to synchronize "+
-						"wallet to chain: %v", err))
+				t := time.Duration(0)
+			out:
+				for {
+					select {
+					case <-time.After(t):
+						t = time.Minute
+						// Sync may be interrupted by actions
+						// such as locking the wallet. Try
+						// again after waiting a bit.
+						err = w.syncWithChain(birthdayBlock)
+						if err != nil {
+							if w.ShuttingDown() {
+								return
+							}
+							log.Errorf("unable to synchronize "+
+								"wallet to chain, trying again "+
+								"in one minute: %v", err)
+							break
+						}
+						break out
+					case <-w.quitChan():
+						return
+					}
 				}
```

Comment:
Very dumb loop, but I think better than the alternative of blowing up. In the case of the wallet being locked, it loops until unlocked. 

However, testing on testnet, I sometimes see an error like `022-12-09 21:22:38.865 [ERR] BTCW: unable to synchronize wallet to chain, trying again in one minute: unable to persist initial sync data: failed to store sync information 00000000000000d47ad9520e3aff5563601586cb4e19ede996827839e03a8326: failed to fetch block hash for height 1998551: block not found` after forcing a loop with a wallet lock.

Unsure atm if that is a result of these changes.